test/greenlight/test_test.clj

```diff
@@ -47,10 +49,35 @@
 (deftest optional-attr-map-test
   (let [system (component/system-map :greenlight.test-test/component 6)
         attr-test (blue/optional-attr-map)
-        test-result (test/run-test! system attr-test)]
+        test-result (test/run-test! system {} attr-test)]
     (is (= (::test/description test-result) "foobar"))
     (is (= (::test/context test-result) {:foo :bar}))
     (is (= :pass (::test/outcome test-result)))
     (is (= ["step-1"]
            (mapv ::step/title (::test/steps test-result))))
     (is (= 1 (count (::test/steps test-result))))))
+
+
+(defmacro with-io
+  [input & forms]
+  `(binding [*in* (io/reader (char-array ~input))
+             *out* (io/writer (java.io.ByteArrayOutputStream.))]
```

Comment:
Is this just to suppress standard output? There's `with-out-str` if you don't care about the return value of `~@forms`. If you want to keep this style then similar to the above, you could just make this a `java.io.StringWriter` directly.

Reply:
`with-out-str` eats the return value, which is used here.

Reply:
Oh you totally mentioned that. 👍